vspace/src/main/java/edu/asu/diging/vspace/core/data/SlideRepository.java

```diff
@@ -19,5 +22,11 @@
     
     @Query("SELECT d.sequence FROM Slide d WHERE d.id = ?1")
     public List<Sequence> getSequencesForSlide(String slideId);
-
+    
+    @Query("SELECT d from Slide d WHERE d.name like %?1% OR d.description like %?1%")
+    public Page<IVSpaceElement> findInNameOrDescription(Pageable requestedPage,String searchText);
+    
+    @Query("SELECT COUNT(distinct d.id) from Slide d WHERE d.name like %?1% OR d.description like %?1%")
+    public long findInNameOrDescription(String searchText);
```

Comment:
since these are pretty simple queries,  just use the automatic query generation for the above methods:
https://docs.spring.io/spring-data/jpa/docs/current/reference/html/#jpa.query-methods

especially since the name of the second method doesn't reflect what it actually does (counting).